spec/models/declaration_spec.rb

```diff
@@ -203,7 +203,7 @@
         completed_declaration.dup.update!(declaration_type: "retained-1")
 
         # Declaration on another provider.
-        completed_declaration.dup.update!(lead_provider: create(:lead_provider, name: "Other lead provider"))
+        completed_declaration.dup.update!(lead_provider: LeadProvider.where.not(id: lead_provider.id).first)
```

Comment:
I've seen this change multiple times, can I ask why we are changing this?

Reply:
Hi @cwrw you ok? Still getting flakey tests when doing `create(:lead_provider, name: "Other lead provider")`.. and then no flakey tests with this change.. hope it's all good. Tks.